### ISSUE TITLE
iVGA input for CasparCG

### ISSUE THREAD

**Issue by TKooijmans** [CasparCG/server#379]
It would be nice to have a i-VGA input for CasperCG.
Newtek has the i-VGA app free for download with makes it possible to use any PC screen in the
network as video input for Tri-Caster.
This works as a video and audio over IP transmitter.
If this can work as input for CasperCG we do not need extra hardware for most external sources.

**Comment by jesperstarkar** [comment]
Newteks new NDI protocol will be a better solution to IP infrastructure.

**Comment by aircooled76** [comment]
+1 for this!

**Comment by aircooled76** [comment]
Would it work to make a VNC based input to achive this?

**Comment by TKooijmans** [comment]
vnc is not fast enough!
I saw NDI on tricaster on IBC and that works as fast as a normal SDI input!

**Comment by aircooled76** [comment]
Just eead up on NDI... it is awesome... Would be great to build an NDI consumer and producer... very powerfull stuff!

**Comment by Keksstar** [comment]
+1

**Comment by hummelstrand** [comment]
The AirSend updater is supposed to update your system to be able to at least send over NDI, and forum users have reported that it works fine with CasparCG Server: http://pages.newtek.com/NDI-Upgrader-Download.html

**Comment by mjbartholomew** [comment]
Any idea on the undertaking to implement an NDI consumer and producer?

**Comment by krzyc** [comment]
How open is this NewTek's solution? Is it tied to Microsoft Windows as stated in minimum system requirements? Are there any limitations when using this protocol outside NewTek's ecosystem? How it is licensed?

I have applied for SDK access so I will probably get answers soon.

**Comment by mjbartholomew** [comment]
I have not attempted to gain SDK access, but I do know that Sienna has NDI apps for the iPad, iPhone and Mac OS X.

http://www.sienna-tv.com/ndi/

**Comment by aircooled76** [comment]
vMix also has free NDI screen capture apps for windows and OSX at vmix.com

Very cool!

**Comment by Keksstar** [comment]
html,body{font-size:12px;background-color:#fff;color:#333;font-family:sans-serif,Arial,Verdana,"Trebuchet MS";line-height:1.4;}I´m using the NDIMonitor direclty from Newtek as an Consumer. Works pretty well with the iVGA Interface from Caspar.

The way back in is missing :-/



aircooled76 schrieb am 17.04.2016 17:51:

vMix also has free NDI screen capture apps for windows and OSX at vmix.com

Very cool!

**Comment by Keksstar** [comment]
If someone need it, I could provide the SDK to build an NDI producer. Would be great if someone could implement it :-)

Consumer doesn´t need to be builded in my eyes, cause the iVGA output is already working and works fine with the NDI AirSend Updater.

**Comment by HellGore** [comment]
It is in my TODO list and I already have the SDK. And by the way anyone can get the NDI SDK from NewTek.

**Comment by dodgepong** [comment]
Forgive my ignorance, but CasparCG is a GPL3 application, right? How would the distribution of an NDI producer work under under GPL3? It is my understanding that for GPL compatibility, NDI library/header sources would need to be distributable, and I don't believe the NDI SDK license allows for the distribution of their source files. Is there a plan to use some sort of inter-process communication to read NDI sources?

I ask because I am in a similar situation with a GPL program that wants to be able to take NDI inputs, and I am curious as to how CasparCG plans on solving this issue.

**Comment by didikunz** [comment]
Any news on this.

**Comment by HellGore** [comment]
I think we would do it in the same way as with the existing iVGA-support:

* Replicate the signature of the functions you use from the NewTek headers and use dlsym/GetProcAddress to set the function pointers at runtime.
* Don't require the closed source .so/.dll to be installed for your product to work (a failed dlopen/LoadLIbrary should not cause you to fail your program), thus allowing a user to go only open source or allow a little closed source.
* Consider whether you think the library goes under the "System Library Exception" in the GPLv3 (like decklink and bluefish probably does) or if you have to make a special exception for the library (https://www.gnu.org/licenses/gpl-faq.html#FSWithNFLibs).

I don't remember if AirSend (iVGA) was considered a System Library or if we did a special exception for it, but anyway the reasoning should be the same with the full NDI SDK IMHO.

**Comment by jesperstarkar** [comment]
Thank you for the brilliant answer. Good to know!

**Comment by didikunz** [comment]
Aha, I was not concerned about the license issue, that is good to know. Just wanted to know, if the NDI input functionality is coming soon or not. Sorry of not expressing myself clearly. :)

**Comment by HellGore** [comment]
@didikunz yes, my answer was actually only in reply to the question asked by @dodgepong. You must have written your post while I was writing mine.

I have some code on my disk regarding this just to get a feel for the SDK but I have not spent time on it lately.

**Comment by TKooijmans** [comment]
Newtek has a tool called Transmit.
With this tool any NDI source, desktop or window can be resend too a virtual Video and audio device.
This virtual device can be seen as a capture device like a webcam by Skype etc.
I have not tried this yet, but I think it's also possible to get it as input device for CasperCG

**Comment by sesse** [comment]
FWIW, FSF's position on proprietary plugins in GPL software is that they must run in the same process (https://www.gnu.org/licenses/gpl-faq.html#GPLAndPlugins). This specifically means that dlopen/GetProcAddress tricks do not change anything, and that GPLv3 software cannot link to NDI. These issues are murky, and what matters is what a judge would rule in court, but one could at least assume that as license writers, their interpretation would be heard and possibly given some weight. GPLv3 is somewhat clearer on this than GPLv2 was; the text now reads:

> For example, Corresponding Source includes interface definition files associated with source files for the work, and the source code for shared libraries and dynamically linked subprograms that the work is specifically designed to require, such as by intimate data communication or control flow between those subprograms and other parts of the work.

You could discuss what “specifically meant to require” means, but again, FSF's FAQ here is clear on their own interpretation.

As for the other possibilities, NDI is obviously not a system library in the context of GPL. Again, the GPLv3 is clear on what's intended:

> The “System Libraries” of an executable work include anything, other than the work as a whole, that (a) is included in the normal form of packaging a Major Component, but which is not part of that Major Component, and (b) serves only to enable use of the work with that Major Component, or to implement a Standard Interface for which an implementation is available to the public in source code form. A “Major Component”, in this context, means a major essential component (kernel, window system, and so on) of the specific operating system (if any) on which the executable work runs, or a compiler used to produce the work, or an object code interpreter used to run it.

NDI is definitely no essential component of Windows or Linux (it's not even distributed with either), so a) falls.

The standard way of dealing with this is to grant a license exception (this is common for code that uses OpenSSL, and the GPLv3 section 7 explains how), but you would need to get every copyright holder of CasparCG (ie., everyone who has ever contributed any code, unless they have done copyright assignment) and any GPL code it uses, to agree to such an exception—one copyright holder cannot declare this unilaterally.

So my advice as a layman would be to make a producer that can publish to a pipe or shared memory segment under some defined protocol, and then have a separate NDI program that can read data from this protocol. (Similarly for consumers.) I understand this is inconvenient, but licenses don't really bow to wishful thinking. :-)

**Comment by krzyc** [comment]
Easiest option is to ask Newtek to relicense NDI SDK to GPLv2.
Also I wonder if they already use some GPLv2 code in their libraries as it is common practice to abuse OSS licenses in commercial projects - there are not much success stories on fighting with these practises unfortunately.

But probably best option is to write GPLv2 compatible implementation of NDI - I do not know whether there will be patent issues or something.

By the way - are there same license issues for using Adobe, Bluefish and Decklink libraries by CasparCG?

**Comment by dodgepong** [comment]
NewTek recently released their headers under the MIT license, and made redistributables available for download from their site, which could possibly work if users want to take the extra step of downloading extra files from NewTek to enable NDI functionality.

**Comment by sesse** [comment]
NewTek has been asked to relicense their NDI SDK before, and has refused on the grounds of interoptability. Note that CasparCG is GPLv3, not GPLv2, so you'd need GPLv2+ or GPLv3(+) if so.

If people do the combination of CasparCG and non-free stuff themselves, that's fine, but the result becomes nondistributable (so you can use it yourself, but you can't e.g. sell a solution that's based on it). As for the status of Adobe, Bluefish and Decklink libraries, I'm less sure about the details for a variety of reasons, so I'll refrain from commenting.

**Comment by RuneFog** [comment]
In the latest SDK, NewTek has made a redistributable dynamic loadable library accessible:

DYNAMIC LOADING OF NDI LIBRARIES 
It is common that you do not want to link directly against the NDI libraries, but instead want to dynamically load them at run-time (which is of particular value in Open Source projects). There is a structure that contains all of the NDI entry points for a particular SDK version, and you can call a single entry point in the library to recover all of these functions. This would allow you on Windows for instance to use LoadLibrary to load the NDI DLL file and GetProcAddress to locate the single entry point. 
NDIlib_v2* p_NDILib = NDIlib_v2_load(); 
At this point, every NDI function has an equivalent member within the NDIlib_v2 structure. For instance, to initialize a sender you can replace a call to NDIlib_find_create2 in the following way: 
NDIlib_find_create2( … ) becomes  p_NDILib->NDIlib_find_create2( … ) 
The preferred mechanism is that you include the NDI DLLs within your application install, but if you do not wish or cannot2 to distribute these then you can ask users to install the NDI redistributable that is both included within the SDK or available as a download from NewTek at http://new.tk/NDIRedistV2 .  
When installed, the location of the redistributable can be found through the environment variable NDI_RUNTIME_DIR_V2. A full example is provided in the NDIlib_DynamicLoad that is included with the SDK and illustrates how to dynamically load the NDI libraries if they are available.

**Comment by walterav1984** [comment]
@RuneFog 
I can confirm this Dynamic Loading Of NDI Libraries section is also listed in the linux NewTek NDI SDK version 2017-02-01 @ r73205.

However the URL to download the NDI redistributable http://new.tk/NDIRedistV2 is serving a MS Windows binary instead of a Linux one.

@HellGore 
Did you also receive the Newtek NDI SDK for linux besides the Windows SDK, it took me a different email address and a month of waiting before Newtek applied.

```
md5 9d1ebc495f603989cf0061a3d761b4f2 "NewTek NDI SDK (Linux).zip" #r72018
md5 0268b8dccaa662f6c043f47d73488db4 "NewTek NDI SDK (Linux).zip" #r73205
```

**Comment by diegofromparis** [comment]
+1

Important feature for CasparCG 2.1

**Comment by nicoaguero** [comment]
I add to the request that CasparCG Server 2.1 contains NDI for input, because it is something that would really be of great use to those who use CasparCG for their productions.

**Comment by didikunz** [comment]
As I can see from reading trough NDI's SDK documentation, there is a way to dynamically link in all neccessary dll's. That is made for all open source projects, that work with a different license, so that the user can download an install it. We use this mechanism alreday for sending out NDI. We can also use it for receiving NDI. So I see no reason, why it should give any licensing issues and would ask for integration of this feature asap, as I need it badly :)

**Comment by vimlesh1975** [comment]
Yes. I also want to test this new feature if implemented.

**Comment by sesse** [comment]
> As I can see from reading trough NDI's SDK documentation, there is a way to dynamically link in all neccessary dll's. […] So I see no reason, why it should give any licensing issues

See my comment on January 7th. Specifically the part starting with: “Loading runtime does not matter much for the license”

**Comment by didikunz** [comment]
We should stop the license discussion here, as if takes all these license terms by the word, we could stop CasparCG from beeing open source as also the BMD drivers are not open source etc. NewTek every year shows CasparCG's logo on their booth at IBC and so they want us to use their technology regardles of the exact terms. Also Flash player would need to be remove. Don't let us be "more catholic than the pope".

**Comment by sesse** [comment]
> NewTek every year shows CasparCG's logo on their booth at IBC

CasparCG contains GPL code, and NewTek is in no position to excuse CasparCG from the GPL.

**Comment by didikunz** [comment]
> HEADER FILES. (NDI_SDK_DIR\INCLUDE\*.H)
These files may be distributed with open source projects under the MIT license. These headers may be included in open source projects (see “Dynamic Loading” section for preferred mechanism), however the requirements of these projects in terms of visual identification of NDI shall be as outlined within the License section above.

**Comment by RuneFog** [comment]
Didi, newtek isn't the problem. 
The problem is that CasparCG has been released under a restrictive license that does not allow linking against closed sourced libraries unless, in sesses interpretation, anybody who has ever contributed signs an exception. 
This might be the letter of the gpl, but I'm not sure It's the spirit. If the developer base had been much smaller and closed group, it would be pretty easy to write the exception.

**Comment by RuneFog** [comment]
Actually, it seems that SVT is still copyright holder on (most of) the files.

**Comment by sesse** [comment]
> Actually, it seems that SVT is still copyright holder on (most of) the files.

You don't need to write “Copyright <your name>” to have copyright on a work; you get it automatically (since the Berne Convention in 1886). So even if the header says “Copyright SVT”, there will be a shared copyright between them and whoever else contributed to that file (even if they didn't add their own name), given that there hasn't been an explicit copyright assignment in place.

**Comment by RuneFog** [comment]
I still fail to see the difference between the decklink sdk and the ndi sdk. They're both MIT-like license APIs with propritary blobs. They both have the blobs available as separate downloads. As didi pointed out, the strict interpretation of the license would require removal of bmd support in casparcg.

**Comment by berge** [comment]
It's not entirely unreasonable (from a licence interpretation perspective) to argue that the BMD driver support in Caspar is out of compliance with GPLv3. The licence does allow linking against non-free system libraries that are "Major Components", but that definition is quite strict and excludes things like third-party drivers. Specifically (from section 1 of GPLv3):

>  A “Major Component”, in this context, means a major essential component (kernel, window system, and so on) of the specific operating system (if any) on which the executable work runs, or a compiler used to produce the work, or an object code interpreter used to run it.

BMD drivers are not distributed with Windows, and are hardly an essential component of the operating system. The GPL FAQ [is pretty explicit on this point](https://www.gnu.org/licenses/gpl-faq.html#SystemLibraryException). To allow linking against non-free system modules, an [exception must be granted in the relevant files](https://www.gnu.org/licenses/gpl-faq.en.html#LinkingOverControlledInterface).

In the case of Caspar, I also think it's fair to argue that the author's intentions most likely was to allow this sort of usage, even though it's not explicitly in the licence. This probably doesn't impact anything, except – and I'm guessing wildly – in the event the project wants to add exceptions to the licence and can't get hold of every copyright holder. _Maybe_ one can just assume that copyright holders for minor contributions would agree to add an exception. There may be precedent here, I don't know.

@RuneFog wrote:
> I still fail to see the difference between the decklink sdk and the ndi sdk. They're both MIT-like license APIs with propritary blobs.

In the context of the GPL, I agree. However, personally I do think there's a distinct difference between the two: The decklink drivers are _drivers_ in a very traditional and obvious sense of the word: They provide a software interface for hardware units, they do essentially nothing but providing that interface, and the hardware is virtually useless without. The NDI SDK, on the other hand, is not a driver; it's a software library that encodes and decodes video, sends it over the network (and other things), and it's useful without specific hardware.

(There's also an exception in the GPL for usage of (and thus linking against components that offer) interfaces that are official standards ("A “Standard Interface” means an interface that either is an official standard defined by a recognized standards body[..]"). Such an example would be libc or OpenGL, so it's not relevant to this discussion.)

**Comment by jesperstarkar** [comment]
I only see 13 contributors listed here on Github. It might be misleading, coming from the SVN repo before migrating to Github. But if 13 is representable, a change of license driven by copyright holders should be doable.

**Comment by berge** [comment]
A change of licensing terms seems the right way forward.

13 contributors looks like the right ballpark. The git history goes back to 2010, and commits from 2010 to 2013 seems to have been imported from svn, including author information. Somewhat normalized (with a `.mailmap` file see below), git shortlog says:
```
  1924  Robert Nagy
   633  Helge Norberg
    68  Niklas P. Andersson
    25  James Wise
    18  Zebiolo
     9  Walter Sonius
     8  Dimitry Ishenko
     6  krzyc
     5  Jonas Hummelstrand
     4  Antonio Ruano Cuesta
     3  Jesper Stærkær
     3  TK3
     2  Peter Keuter
     2  drakmor
     1  Ovidijus Striaukas
```
…of which at least seven authors' contributions are so trivial I doubt they're copyrightable (and would at any rate be trivial to re-implement).

I guess the interesting bit is the first commit, from 2010-09-14. It's 13k SLOC, commited by @ronag. It does not contain any licensing information (the LICENCE file was added in 2013), and it's not clear if he holds the copyright to all of that code. I don't know the project's early history; was it originally an internal project at SVT, later open sourced?

If the project decides to relicense, I'd suggest 1) someone comes up with a proper license suggestion (GPLv3 plus exceptions, for instance), 2) goes through the list of dependencies to make sure Caspar doesn't link against any pure-GPL-code (since GPL + exceptions in incompatible with pure GPL) and 3) _then_ contacts all the copyright holders with a rationale and the suggestion.

`.mailmap` file for normalizing contributions:
```
Robert Nagy <ronag@362d55ac-95cf-4e76-9f9a-cbaa9c17b72d>
Helge Norberg <helge.norberg@gmail.com>
Helge Norberg <hellgore@362d55ac-95cf-4e76-9f9a-cbaa9c17b72d>
Niklas P. Andersson <niklas.p.andersson@svt.se>
Walter Sonius <walterav1984@gmail.com>
TK3 <tk-3@362d55ac-95cf-4e76-9f9a-cbaa9c17b72d>
Peter Keuter <Peter@Peter-PC.(none)>```

**Comment by jesperstarkar** [comment]
ping @dotarmin

**Comment by didikunz** [comment]
CasparCG was a internal project before beeing open sourced. AFAIK Rober Nagy was the laed developer at that time. So as an employee it's normally the case, that your copyright goes to the company, that pays you. That is normally part of the contract. So for these early days I think SVT is the sole copyright holder. Of course that does not cover the contributions Robert made (and still do), after he left SVT and was a freelance developer.

**Comment by jesperstarkar** [comment]
Robert was not the initial developer, but everything prior to 2010 is SVT internally.

@berge Would a full blown permissive license be an option, if SVT want to go down that road. Could MIT fly?

**Comment by berge** [comment]
I'm just a random user with zero contributions to the project, so my opinion on this matters very little (-: However, I'd strongly recommend against a non-copyleft-license such as MIT or BSD. The graphics playout ecosystem seems to have few and mostly non-free offerings, often at considerable expense to users. With a non-copyleft license, there's a very real risk that Caspar will be forked to a (or several) non-free project(s), which definitely would not help the main project.

**Comment by jesperstarkar** [comment]
We have only discussed BMD and NDI for exceptions, but perhaps Flash player should be thrown in too. How about CEF (BSD)?

**Comment by didikunz** [comment]
> We have only discussed BMD and NDI for exceptions, but perhaps Flash player should be thrown in too. How about CEF (BSD)?

It starts to becoume confusing :)

**Comment by sesse** [comment]
> We have only discussed BMD and NDI for exceptions, but perhaps Flash player should be thrown in too. How about CEF (BSD)?

The 2-clause BSD license is pretty much compatible with everything, so you don't need an exception for it.

However, if you change from GPLv3 to GPLv3 with exception clauses, you are no longer GPLv3 (nor GPLv2) compatible, and cannot link to GPL libraries unless they too include such exceptions. In particular, the FFmpeg compile CasparCG is shipping is compiled with --enable-gpl --enable-v3, and thus GPLv3, so that needs to change. Removing --enable-gpl in turn disables e.g. x264 support, since x264 is licensed under the GPL.

**Comment by dotarmin** [comment]
@jesperstarkar Before I start commenting this issue I need to spend time on reading about the various licenses etc. I think it is really interesting that this issue has got attention again! 

The comments and discussions so far has been interesting, I will spend little time to read more about various licenses etc.to get involved into this discussion.

**Comment by jamesdwise** [comment]
Don't forget the Bluefish Component too.

However as the developer behind some of that code, and an employee @ Bluefish444,
I can assure you that use of the code and linking to the driver is perfectly fine with Bluefish and exactly what they want.

To be honest i am still not 100% on the exact legal requirements either.
Bluefish sells hardware, and the drivers are freely available via the web.
Use of a Bluefish device with Caspar requires having the appropriate driver installed, but that is all.
The Caspar project includes some bluefish header files, which we are totally fine for those to be re-distributed.

Anyway given the current discussion i thought i would jump in and give my 2c - especially since I (sort-of) represent one of the 3rd parties here.

 - James Wise.


** I am not a lawyer, just a programmer.




On 25/08/2017 1:47 am, Steinar H. Gunderson wrote:

We have only discussed BMD and NDI for exceptions, but perhaps Flash player should be thrown in too. How about CEF (BSD)?

The 2-clause BSD license is pretty much compatible with everything, so you don't need an exception for it.

However, if you change from GPLv3 to GPLv3 with exception clauses, you are no longer GPLv3 (nor GPLv2) compatible, and cannot link to GPL libraries unless they too include such exceptions. In particular, the FFmpeg compile CasparCG is shipping is compiled with --enable-gpl --enable-v3, and thus GPLv3, so that needs to change. Removing --enable-gpl in turn disables e.g. x264 support, since x264 is licensed under the GPL.

-

**Comment by Keksstar** [comment]
Any update what happend in the background so far?
Is the NDI Input still possible or has licensing to be clarified before that?

**Comment by calumbrookes** [comment]
+1 to this, definitely interested in helping put man-hours in towards getting NDI into Caspar if I can be of use to the project.

**Comment by aircooled76** [comment]
If the use of other components are questionable as mentioned here
(BlackMagic etc.) Should we work to build the code for NDI Input first and
work through the licence implications after?

This would be an awesome addition to CasperCG and would enable me to expand
my use of the server...

On 1 Nov. 2017 03:45, "Calum Brookes" <notifications@github.com> wrote:

+1 to this, definitely interested in helping put man-hours in towards
getting NDI into Caspar if I can be of use to the project.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/CasparCG/Server/issues/379#issuecomment-340835319>, or mute
the thread

**Comment by calumbrookes** [comment]
OK so some digging later...

This [FFmpeg commit](https://github.com/FFmpeg/FFmpeg/commit/2634927fe30ea4a821db515c6b7f77458f5c4bc5) shows that FFmpeg 3.4 includes NDI support. If this is the case, then is it possible via this method to get NDI into Caspar by updating the FFmpeg libraries to 3.4 and using the FFmpeg producer and using an ndi:// URL rather than adding a separate NDI producer?

**Comment by Palakis** [comment]
@calumbrookes No it doesn't circumvent the licensing issues, because it would mean that the program relies on a proprietary component to function and the GPL tries to avoid that.

**Comment by Palakis** [comment]
By looking at the source of ndi ffmpeg module, it looks like it does compile-time linking to the library. IIRC doing so prevents FFmpeg from running without the NDI library installed, and thus requires bundling these with FFmpeg.

Another solution would be to use dynamic loading. NewTek provides such a mechanism in their MIT-licensed NDI headers provided in the SDK.

**Comment by Julusian** [comment]
I have been thinking about this problem quite a bit lately. After doing a load of research on various aspects and considering whether we could change to another license, I have a conclusion that I think is reasonable. So far I have only been thinking in terms of the NDI lib, but I expect the same can be extended to decklink and others.

First off, we have to stay with the GPL. We link against ffmpeg and x264, which is GPL, so we are required to be GPL in order to do that. Even if we were to change to LGPL or MIT, as soon as we link to ffmpeg we would be classed as GPL so nothing would have changed. 
We can't add any library exceptions to our GPL as doing so would also mean that we cant link against any other GPL libraries, so again would not work for us. This means we need to find a way to make this work while remaining under GPL.

The issue we have with the GPL is that it doesn't allow us to link against proprietary libraries. However, as far as I am aware that is only an issue if the libs are required to run the program or are compile time linked. The user is allowed to combine a GPL program with non-GPL compatible plugins/modules for their use, but they are not allowed to distribute the resulting work.

If using NDI via the dynamic loading method, then the only new code we will introduce will be an MIT header, which is compatible with the GPL. So the resulting build of caspar will still be GPL compliant. It is only once the user combines it with the NDI runtime that the result will no longer be GPL compliant, but that only stops them distributing. They are still allowed to use it themselves.

I think we can ignore the bits of the license saying it needs to be in its own process/program, as that is only required if we want to be able to ship the NDI runtime with caspar. Ultimately GPL violations are fine if you only use it yourself, unless you are distributing the result.


TLDR;
I believe we can use the NDI MIT headers, and ask the user to install the optional NDI runtime themselves if they wish without the project violating GPL.

I'm interested to know what others think of this.

**Comment by Palakis** [comment]
> TLDR;
I believe we can use the NDI MIT headers, and ask the user to install the optional NDI runtime themselves if they wish without the project violating GPL.

I definitely second you on this!

**Comment by ahallez** [comment]
For what it's worth, OBS as well as it's NDI integration plugin have a GPL v2 license.
Both are available even if the discussion about the license compatibility is still open:
https://github.com/Palakis/obs-ndi/issues/230

**Comment by Julusian** [comment]
A new NDI producer and consumer has been merged for 2.3 https://github.com/CasparCG/server/pull/1112
